testutils/backend/db/testing/repository.go

```diff
@@ -0,0 +1,125 @@
+// Code generated by mockery v1.0.0. DO NOT EDIT.
+
+package mocks
+
+import (
+	context "context"
+	model "github.com/opensds/multi-cloud/backend/pkg/model"
+	mock "github.com/stretchr/testify/mock"
+)
+
+// Repository is an autogenerated mock type for the Repository type
+type Repository struct {
+	mock.Mock
+}
+
+// Close provides a mock function with given fields:
+func (_m *Repository) Close() {
+	_m.Called()
+}
+
+// CreateBackend provides a mock function with given fields: ctx, backend
+func (_m *Repository) CreateBackend(ctx context.Context, backend *model.Backend) (*model.Backend, error) {
+	ret := _m.Called(ctx, backend)
+
+	var r0 *model.Backend
+	if rf, ok := ret.Get(0).(func(context.Context, *model.Backend) *model.Backend); ok {
+		r0 = rf(ctx, backend)
+	} else {
+		if ret.Get(0) != nil {
+			r0 = ret.Get(0).(*model.Backend)
+		}
+	}
+
+	var r1 error
+	if rf, ok := ret.Get(1).(func(context.Context, *model.Backend) error); ok {
+		r1 = rf(ctx, backend)
+	} else {
+		r1 = ret.Error(1)
+	}
+
+	return r0, r1
+}
+
+// DeleteBackend provides a mock function with given fields: ctx, id
+func (_m *Repository) DeleteBackend(ctx context.Context, id string) error {
+	ret := _m.Called(ctx, id)
+
+	var r0 error
+	if rf, ok := ret.Get(0).(func(context.Context, string) error); ok {
+		r0 = rf(ctx, id)
+	} else {
+		r0 = ret.Error(0)
+	}
+
+	return r0
+}
+
+// GetBackend provides a mock function with given fields: ctx, id
+func (_m *Repository) GetBackend(ctx context.Context, id string) (*model.Backend, error) {
+	ret := _m.Called(ctx, id)
+
+	var r0 *model.Backend
+	if rf, ok := ret.Get(0).(func(context.Context, string) *model.Backend); ok {
+		r0 = rf(ctx, id)
+	} else {
+		if ret.Get(0) != nil {
+			r0 = ret.Get(0).(*model.Backend)
+		}
+	}
+
+	var r1 error
+	if rf, ok := ret.Get(1).(func(context.Context, string) error); ok {
+		r1 = rf(ctx, id)
+	} else {
+		r1 = ret.Error(1)
+	}
+
+	return r0, r1
+}
+
+// ListBackend provides a mock function with given fields: ctx, limit, offset, query
+func (_m *Repository) ListBackend(ctx context.Context, limit int, offset int, query interface{}) ([]*model.Backend, error) {
+	ret := _m.Called(ctx, limit, offset, query)
+
+	var r0 []*model.Backend
+	if rf, ok := ret.Get(0).(func(context.Context, int, int, interface{}) []*model.Backend); ok {
```

Comment:
Is this specific to length of data we provide. I mean right now the list has just two elements so it will call Get(0) and Get(1)
But what if the data size increases. Can we write generic Get(i) based upon the length